### DISCUSSION TITLE
Is persisting data for rabbitmq container not a good idea?

### DISCUSSION THREAD

**Discussion by nicholasyin** [rabbitmq/rabbitmq-server#3871] [Questions] [answered]
I'm asking because I don't find any official instruction on how to persist data for rabbitmq running in a docker container, how to mount custom volume/directory, etc. This makes me wondering if this is not a good idea to persist data across container construction and destruction?

**Comment by mkuratczyk** [comment]
Hi.

Almost every production RabbitMQ deployment uses persistent data. However,
it can be a bad idea to use a custom Docker-based deployments for managing
stateful applications in general. If you are trying to deploy a production
environment, and want to use containers, then you probably should use
Kubernetes and the RabbitMQ Operator:
https://www.rabbitmq.com/kubernetes/operator/operator-overview.html.
If, for some reason, you can't/don't want to, then it can still serve as an
example of how to deploy the RabbitMQ image, what volumes to mount, etc.

Best,

On Thu, Dec 9, 2021 at 1:37 AM Nico ***@***.***> wrote:

> I'm asking because I don't find any official instruction on how to persist
> data for rabbitmq running in a docker container, how to mount custom
> volume/directory, etc. This makes me wondering if this is not a good idea
> to persist data across container construction and destruction?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-server/discussions/3871>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACI7PIQUYD35MYVKXUQGN2TUP726NANCNFSM5JVD4H4Q>
> .
>


-- 
Michał
RabbitMQ team

**Reply by nicholasyin** [reply]
Sorry for not mentioning that I'm using docker compose to set up a dev environment, so would like to try to keep away from Kubernetes as much as I can to make the dev enrivonment setup simpler. Is there any document regarding docker compose scenario?

**Comment by mkuratczyk** [comment, marked as answer]
For a dev env, not keeping persisting data is likely what you want anyway -
just start fresh every time.
We don't have an official guide for docker compose. We do use it for some
testing so perhaps it can serve as an example:
https://github.com/rabbitmq/rabbitmq-server/tree/master/deps/rabbitmq_prometheus/docker
.

Personally I have virtually zero experience with Docker Compose so I won't
be able to assist with any details, but hopefully someone can help. I'd
just deploy a local Kubernetes (kind, minikube or something), install the
operator and I'd have RabbitMQ working within a minute or so.

On Thu, Dec 9, 2021 at 12:29 PM Nico ***@***.***> wrote:

> Sorry for not mentioning that I'm using docker compose to set up a dev
> environment, so would like to try to keep away from Kubernetes as much as I
> can to make the dev enrivonment setup simpler. Is there any document
> regarding docker compose scenario?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-server/discussions/3871#discussioncomment-1778443>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACI7PIQACQNJ45OLP2KDHHLUQCHLBANCNFSM5JVD4H4Q>
> .
>


-- 
Michał

**Comment by MAYousuf** [comment]
You can check below link
https://www.futurefundamentals.com/set-rabbitmq-with-data-persistent-with-docker-compose/

**Reply by nicholasyin** [reply]
Good to know. Thanks.
